core/src/main/java/com/google/instrumentation/trace/ProbabilitySampler.java

```diff
@@ -0,0 +1,92 @@
+package com.google.instrumentation.trace;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.math.BigInteger;
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * Make decisions on sampling based on a fixed probability. Spans will be sampled if either:
+ *
+ * <ul>
+ *   <li>the trace ID falls within a range proportional to the probability; OR
+ *   <li>they are a child span, and parent was sampled.
+ * </ul>
+ */
+public final class ProbabilitySampler extends Sampler {
+  // We assume the lower 64 bits of the traceId's are randomly distributed around the whole (long)
+  // range. We convert an incoming probability into an upper bound on that value, such that we can
+  // just compare the absolute value of the id and the bound to see if we are within the desired
+  // probability range.  Using the low bits of the traceId also ensures that systems that only use
+  // 64 bit ID's will also work with this sampler.
+  private final long idUpperBound;
+  private final double probability;
+
+  /**
+   * Create a new {@link ProbabilitySampler}. The probability of sampling a trace is equal to that
+   * of the specified probability.
+   *
+   * @param probability The desired probability of sampling. Must be within [0.0, 1.0].
+   * @throws IllegalArgumentException if {@code probability} is out of range
+   */
+  public ProbabilitySampler(double probability) {
```

Comment:
Please add a factory method in the Samplers:

public static Sampler getProbabilitySampler(double probability) {
}

And make the ctor and the class package protected. Consider also to move this into Sampler as a private class.